tests/src/core/users/DetermineLifetimeConditions.test.ts

```diff
@@ -76,7 +76,111 @@ describe('Determining Lifetime conditions', () => {
+      );
+    });
+  });
+
+  describe('The user is free', () => {
+    it('When the user is free, then the tier and the maxSpaceBytes tier field are returned', async () => {
+      const mockedUser = getUser({
+        lifetime: false,
+      });
+      const mockedTier = newTier({
+        billingType: 'lifetime',
+      });
+
+      jest.spyOn(paymentService, 'getUserSubscription').mockResolvedValue({ type: 'free' });
+      jest.spyOn(tiersService, 'getTierProductsByProductsId').mockResolvedValue(mockedTier);
+
+      const { maxSpaceBytes, tier } = await determineLifetimeConditions.determine(mockedUser, mockedTier.productId);
+
+      expect(tier).toStrictEqual(mockedTier);
+      expect(maxSpaceBytes).toStrictEqual(mockedTier.featuresPerService[Service.Drive].maxSpaceBytes);
+    });
+  });
+
+  describe('The user already has a subscription', () => {
+    it('When the user has an active subscription, then the subscription is cancelled and the lifetime tier is returned', async () => {
+      const mockedUser = getUser({
+        lifetime: false,
+      });
+      const mockedUserSubscription = getSubscription({ type: 'subscription' });
+      const subscriptionId = mockedUserSubscription.type === 'subscription' && mockedUserSubscription.subscriptionId;
+      const mockedTier = newTier({
+        billingType: 'lifetime',
+      });
+
+      jest.spyOn(paymentService, 'getUserSubscription').mockResolvedValue(mockedUserSubscription);
+      jest.spyOn(tiersService, 'getTierProductsByProductsId').mockResolvedValue(mockedTier);
+      const cancelSubscriptionSpy = jest.spyOn(paymentService, 'cancelSubscription').mockResolvedValue();
+
+      const { maxSpaceBytes, tier } = await determineLifetimeConditions.determine(mockedUser, mockedTier.productId);
+
+      expect(cancelSubscriptionSpy).toHaveBeenCalledTimes(1);
+      expect(cancelSubscriptionSpy).toHaveBeenCalledWith(subscriptionId as string);
+      expect(tier).toStrictEqual(mockedTier);
+      expect(maxSpaceBytes).toStrictEqual(tier.featuresPerService[Service.Drive].maxSpaceBytes);
+    });
+  });
+
+  describe('The user already has a lifetime plan', () => {
+    it('When the user already has a lifetime, then the storage should be stacked', async () => {
+      const mockedUser = getUser({ lifetime: true });
+      const mockedTier = newTier();
+      jest.spyOn(determineLifetimeConditions as any, 'handleStackingLifetime').mockResolvedValue({
+        tier: mockedTier,
+        maxSpaceBytes: mockedTier.featuresPerService[Service.Drive].maxSpaceBytes,
+      });
+      jest.spyOn(paymentService, 'getUserSubscription').mockResolvedValue({ type: 'free' });
+      jest.spyOn(tiersService, 'getTierProductsByProductsId').mockResolvedValue(mockedTier);
+
+      const { maxSpaceBytes, tier } = await determineLifetimeConditions.determine(mockedUser, mockedTier.productId);
+
+      expect(maxSpaceBytes).toStrictEqual(mockedTier.featuresPerService[Service.Drive].maxSpaceBytes);
```

Comment:
Shouldn't this be `maxSpaceBytes` equals all the storage of all the lifetime tiers stacked?